src/riak_kv_qry_buffers.erl

```diff
@@ -660,6 +719,44 @@ tstamp() ->
 
 
 %% data ops
+
```

Comment:
I would really appreciate a spec attached to this function. The inputs and logic are complex enough, I feel, to warrant it.